docs/csp.md

```diff
@@ -1,36 +1,45 @@
 # Setting up Content Security Policy with JSS
 
-Content Security Policy (CSP) is a way of whitelisting what resources the browser should allow to load (and reach out to). There are many good resouces on CSP for more information. [This explanation](https://helmetjs.github.io/docs/csp/) is a good place to start.
+Content Security Policy (CSP) is a way of whitelisting what resources the browser should allow to load (and reach out to). There are many good resources on CSP for more information. [This explanation](https://helmetjs.github.io/docs/csp/) is a good place to start.
+
+In general you should start with `Content-Security-Policy: default-src 'self';` and enable things as needed from there.
 
 In the case of JSS we need to set the `style-src` CSP directive. We don't want to just set it to `unsafe-inline` which will allow everything.
 
-The solution is to include a _nonce_:
+The solution is to include a _nonce_ (number used once):
 
 [MDN/CSP/style-src](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src) notes the following:
 
 > 'nonce-{base64-value}'
 > A whitelist for specific inline scripts using a cryptographic nonce (number used once). **The server must generate a unique nonce value each time it transmits a policy. It is critical to provide an unguessable nonce, as bypassing a resource’s policy is otherwise trivial**. See [unsafe inline](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src#Unsafe_inline_script) script for an example.
 
-A conundrum for now is how to generate a unique nonce value for every page visit, as webpack must know the nonce at bundle time, while the server must know the nonce value at runtime for including it in the CSP rules. For this tutorial, we'll use a hard-coded Base64-encoded GUID value.
-
-1. Generate a Base64 guid:
+Because the nonce must be generated to be unique and random for every request, this is not something that we can do at compile time. Previously the docs suggested using the `__webpack_nonce__` variable with Webpack. However that is insecure because it never changes, so it could be trivially bypassed by an attacker, as noted in the Mozilla docs above.
 
-   * Get a guid: https://www.guidgenerator.com/online-guid-generator.aspx
-   * Encode the guid: https://www.base64encode.org/
+In order to communicate the nonce value to JSS, we're going use some basic templating with an express server.
 
-1. In server startup:
+1. In the server startup, we'll add middleware to generate the nonce.
 
 ```js
 // server.js
 import helmet from "helmet";
+import uuidv4 from "uuid/v4"
 
-const WEBPACK_NONCE = "N2M0MDhkN2EtMmRkYi00MTExLWFhM2YtNDhkNTc4NGJhMjA3";
-
+app.use(function(req, res, next) {
+  // nonce should be base64 encoded
+  res.locals.nonce = btoa(uuidv4());
+  next();
+});
```

Comment:
lets stick with no-semicolon style) no hard feelings, just consistency

Reply:
I don't really care what format. But formatting should be done by an auto-formatter.
I didn't see anything in place, so I copied the code into https://prettier.io/playground/ and back. If you have a prefered format, please give me an easy way to format the code.

Reply:
havn't started to use prettier yet, still rely on eslint here, but docs are not linted, mb there is  a way to check code blocks in md files by eslint as well?

Reply:
https://github.com/eslint/eslint-plugin-markdown

Reply:
added in master

Reply:
there are plenty of errors now from md files, just lint try to lint the changed file with lint:staged or over commit hook